### PR TITLE
only volume mount explicitly

### PR DESCRIPTION
**Pull request by peterbe**
@milescrabill I think I figured it out. At least it all makes sense locally. 
If you run `docker-compose run test-ci test` it runs the `test-ci` directive from `docker-compose.yml`. So, that'll run with the files "as of the latest `make build`". 

To test this, I do:
```
$ export CI=true
$ make test
```
The tests pass. So I mess with, for example, `tests/test_api.py` and put in a `raise Exception("hi")` and run: `make test` again and the tests pass again. 

However, when I do this WITHOUT setting `export CI=true` the tests will re-read the host's file and the second time the tests fail because of the temporary `raise Exception("hi")` I put in.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -56,8 +56,18 @@ services:
...
     command: test
 
+  test-ci:
+    extends:
+      service: base
+    links:
+      - db
+      - redis-store
+      - redis-cache
+
```
Comment (peterbe) [inline docker-compose.yml]: I tried to be "neater" here to not have to list the `links: ...` two times but docker-compose doesn't allow that. Now I violate the DRY principle but it's fine. 
Note also, that there's no need for a `command` in the `test-ci` directive.

File: pytest.ini [hunk 2]
```diff
@@ -14,4 +15,4 @@ DJANGO_CONFIGURATION = Test
 
 
 [pytest-watch]
-ignore = frontend .git docs test-with-nginx
+ignore = frontend .git docs test-with-nginx .cache htmlcov
```
Comment (peterbe) [inline pytest.ini]: These changes are actually irrelevant. I just noticed it whilst working on this. Sorry.

File: docker-compose.yml (hunk 1 above)
Comment (jgmize) [reply docker-compose.yml]: The explicit links are not necessary: https://docs.docker.com/compose/networking/#links

File: bin/test.sh [hunk 3]
```diff
@@ -8,10 +8,14 @@ set -eo pipefail
 export DEVELOPMENT=1
 export DJANGO_CONFIGURATION=Test
 
-# pass CI env vars into docker containers for codecov submission
-[ ! -z ${CI+check} ] && \
-    echo "Getting Codecov environment variables" && \
+# run docker compose with the given environment variables
+
+if [[ ! -z "${CI}" ]]; then
```
Comment (jgmize) [inline bin/test.sh]: FYI you could replace `! -z` with the equivalent `-n` here

Comment (peterbe) [reply bin/test.sh]: Ohh... I never get these right. I always just google stackoverflow and trial-and-error till it works. It always gets tricky when things are set but set to an empty string.

**Pull request merged**